src/main/java/com/google/sps/servlets/NewRecipeServlet.java

```diff
@@ -46,14 +48,16 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     List<String> ingredients = FormHelper.separateByNewlines(ingredientsResponse);
     List<String> steps = FormHelper.separateByNewlines(stepsResponse);
 
+    User user = new User(idToken);
+
     Entity recipeEntity = new Entity("Recipe");
     recipeEntity.setProperty("name", name);
     recipeEntity.setProperty("tags", tags);
     recipeEntity.setProperty("description", description);
     recipeEntity.setProperty("ingredients", ingredients);
     recipeEntity.setProperty("steps", steps);
+    recipeEntity.setProperty("authorID", user.getId());
```

Comment:
Can you do `recipeEntity.setProperty("authorID", idToken)`? (and not create a User) Or is this id different?

Reply:
the id is different; I have to get it from the idToken
to be honest, I'm just doing what Ali told me; he said to get the id by making a new User